### DISCUSSION TITLE
Trivy.db does not get updated automatically

### DISCUSSION THREAD

**Discussion by pierrferr** [aquasecurity/trivy#4774] [Q&A]
### Question

Hi,
Trivy is running as a container on my Harbor registry VM (trivy-adapter). Vulnerability scanning is working fine.

HTTP_PROXY and HTTPS_PROXY environment variables are configured
I manage to manually update de trivy.db and metadata.json manually (by running "trivy --debug server" for instance). 
But it never get updated automatically every 6 hours as expected

Can you help me to troubleshoot this issue?

Trivy in the harbor docker-compose.yml :

trivy-adapter:
    container_name: trivy-adapter
    image: goharbor/trivy-adapter-photon:v2.6.1
    command: ['chown', '-R', '10000:10000', '/home/scanner/.cache']
    restart: always
    cap_drop:
      - ALL
    depends_on:
      - log
      - redis
    networks:
      - harbor
    volumes:
      - type: bind
        source: /data/trivy-adapter/trivy
        target: /home/scanner/.cache/trivy
      - type: bind
        source: /data/trivy-adapter/reports
        target: /home/scanner/.cache/reports
      - type: bind
        source: ./common/config/shared/trust-certificates
        target: /harbor_cust_cert
    logging:
      driver: "syslog"
      options:
        syslog-address: "tcp://localhost:1514"
        tag: "trivy-adapter"
    env_file:
      ./common/config/trivy-adapter/env

Troubleshooting:  

This might the reason of the update failure :
scanner [ / ]$ trivy image --reset
2023-07-04T13:50:24.866Z        INFO    Removing DB file...
2023-07-04T13:50:24.897Z        FATAL   init error: cache error: cache reset error: failed to clear the database: failed to remove the directory (/home/scanner/.cache/trivy) : unlinkat /home/scanner/.cache/trivy: permission denied

I have added the line " command: ['chown', '-R', '10000:10000', '/home/scanner/.cache']" in the docker-compose.yaml for this reason. Without success.

### Target

None

### Scanner

Vulnerability

### Output Format

None

### Mode

None

### Operating System

RHEL

### Version

```bash
Version: 0.29.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-07-04 12:10:23.757500034 +0000 UTC
  NextUpdate: 2023-07-04 18:10:23.757499534 +0000 UTC
  DownloadedAt: 2023-07-04 13:45:13.955542624 +0000 UTC
```

**Comment by knqyf263** [comment]
@chen-keinan `goharbor/trivy-adapter-photon` is not maintained by us, right?

**Reply by chen-keinan** [reply]
Correct

**Reply by knqyf263** [reply]
@pierrferr You can ask the Harbor team. One thing in my mind is your Trivy version might be too old. Trivy had a problem with proxy, and it was fixed at some point.

I believe it is not our issue. Please let me close this discussion.

**Reply by zyyw** [reply]
`goharbor/trivy-adapter-photon` is an image build from https://github.com/aquasecurity/harbor-scanner-trivy

@pierrferr could you please try the latest Harbor version to see if this issue still persists. If so, we can continue to figure out the reason why it happens.
